src/Xamarin.Android.Build.Tasks/Tasks/AndroidDotnetToolTask.cs

```diff
@@ -46,10 +64,24 @@ public override bool Execute ()
 		}
 
 		protected override string ToolName =>
-			UsingDotnet ? "dotnet" : $"{BaseToolName}.exe";
+			UsingDotnet ? "dotnet" : UsingMono ? "mono" : $"{BaseToolName}.exe";
 
 		protected override string GenerateFullPathToTool () =>
-			UsingDotnet ? "dotnet" : Path.Combine (ToolPath, ToolExe);
+			UsingDotnet ? "dotnet" : UsingMono ? FindMono () : Path.Combine (ToolPath, ToolExe);
```

Comment:
Would it make sense to cache `FindMono ()` result? How often is the task used?

Reply:
I changed this to use `RegisterTaskObject` for a `Build` lifetime.

It could happen a few times per project.